app/controllers/application_controller.rb

```diff
@@ -94,6 +94,12 @@ def migration_error?
     render :migration_error, status: 500 unless ENV["DB_MIGRATE_FAILED"].blank?
   end
 
+  # Determines proper locale to be used by calling user_locale with params based on if room owner exists
+  def determine_locale(room)
+    room.nil? ? user_locale(nil) : user_locale(room.owner)
```

Comment:
I would set use this instead here
```
if user && user.language != 'default'
      user.language
    else
      Rails.configuration.default_locale.presence || http_accept_language.language_region_compatible_from(I18n.available_locales)
    end
```

and then update `user_locale` to use the new method. Then, once you've made that change, you should actually be able to call `determine_locale(@room&.owner)` and that should work because of https://stackoverflow.com/questions/36812647/what-does-ampersand-dot-mean-in-ruby#:~:text=It%20is%20called%20the%20Safe,the%20try%20method%20in%20Rails.&text=This%20sends%20the%20my_method%20message%20to%20my_object%20.